### PR TITLE
Workspace Integration test workflow

### PR DESCRIPTION
**Pull request by orensbruli**
Adds a reusable workflow to test workspaces in Moveit Pro docker containers:
It has been tested in the [moveit_studio_ur_ws](https://github.com/PickNikRobotics/moveit_studio_ur_ws) repo:
https://github.com/PickNikRobotics/moveit_studio_ur_ws/actions/runs/7960078548/job/21728321485

### CODE REVIEW

File: .github/workflows/workspace_integration_test.yaml [hunk 1]
```diff
@@ -1,12 +1,79 @@
...
+      - id: dotenv
+        uses: falti/dotenv-action@v1.0.4
+  integration-test-in-studio-container:
+    needs: setup
+    runs-on: ubuntu-latest
+    env:
+      STUDIO_LICENSE_KEY: ${{ secrets.studio_license_key || secrets.STUDIO_CI_LICENSE_KEY }}
+      # TODO this variable is required but github workspace is different
+      # see https://github.com/actions/runner/issues/2058
+      USER_WS: /github/workspace
```
Comment (sea-bass) [inline .github/workflows/workspace_integration_test.yaml]: Not sure I understand this part.

File: .github/workflows/workspace_integration_test.yaml [hunk 2]
```diff
@@ -1,12 +1,79 @@
...
+      ROS_DOMAIN_ID: 49
+      ROS_LOCALHOST_ONLY: 1
+      RMW_IMPLEMENTATION: rmw_cyclonedds_cpp
+      CYCLONEDDS_URI: /tmp/cyclonedds.xml
+    # TODO: docker tag
+    # TODO: example for custom base images
+    container: picknikciuser/moveit-studio:${{ needs.setup.outputs.image_tag }}
```
Comment (sea-bass) [inline .github/workflows/workspace_integration_test.yaml]: Isn't the docker tag already specified in this line? Is the TODO outdated?

File: .github/workflows/workspace_integration_test.yaml [hunk 3]
```diff
@@ -1,12 +1,79 @@
...
+      - name: Run colcon test
+        run: |
+          . /opt/ros/humble/setup.sh
+          . /opt/underlay_ws/install/setup.sh
+          . /opt/overlay_ws/install/setup.sh
+          . ./install/setup.sh
+          colcon test --retest-until-pass 3
+      - if: always()
+        run: |
+          colcon test-result --verbose
```
Comment (sea-bass) [inline .github/workflows/workspace_integration_test.yaml]: nit: add newline at the end

**Review by sea-bass — COMMENTED** [review]
This looks good!

One higher-level question; should we rename this repo, and thereby action, to something more specific like `moveit_pro_workspace_ci`?

**Comment by orensbruli** [comment]
> One higher-level question; should we rename this repo, and thereby action, to something more specific like moveit_pro_workspace_ci?

The intention is to have a public repo for `Moveit Pro` things, but I guess it's not going to be only related to `workspaces`.

So, maybe, `moveit_pro_ci `or `moveit_pro_reusable_ci`?

File: .github/workflows/workspace_integration_test.yaml (hunk 1 above)
Comment (orensbruli) [reply .github/workflows/workspace_integration_test.yaml]: Me neither :rofl: I mean, this comes from @moriarty code in https://github.com/PickNikRobotics/moveit_studio_ur_ws and I didn't check.
Now, looking into the link I see that there's a confirmed bug related to the paths in the containers in the runners.
I think we could bring some more information about the bug in this comment but keep the hardcoded path.

Comment (orensbruli) [reply .github/workflows/workspace_integration_test.yaml]: 572d25f9e8c5bfa225ed5f644f87bbf737d32dbf

File: .github/workflows/workspace_integration_test.yaml (hunk 2 above)
Comment (orensbruli) [reply .github/workflows/workspace_integration_test.yaml]: yep. thx
d05209c0ad2c67d16e2bed6a9d7493f6628f3430

File: .github/workflows/workspace_integration_test.yaml (hunk 3 above)
Comment (orensbruli) [reply .github/workflows/workspace_integration_test.yaml]: https://github.com/PickNikRobotics/reusable_ci/commit/572d25f9e8c5bfa225ed5f644f87bbf737d32dbf

**Review by sea-bass — APPROVED** [review]
`moveit_pro_ci` seems good!

File: .github/workflows/workspace_integration_test.yaml [hunk 4]
```diff
@@ -1,12 +1,107 @@
...
+      - name: Configure DDS
+        run: |
+          cat <<EOF > /tmp/cyclonedds.xml
+          <?xml version="1.0" ?>
+          <CycloneDDS xmlns="https://cdds.io/config" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://cdds.io/config https://raw.githubusercontent.com/eclipse-cyclonedds/cyclonedds/master/etc/cyclonedds.xsd">
+            <Domain id="any">
+              <Discovery>
+                <ParticipantIndex>auto</ParticipantIndex>
+                <MaxAutoParticipantIndex>120</MaxAutoParticipantIndex>
+              </Discovery>
```
Comment (sea-bass) [inline .github/workflows/workspace_integration_test.yaml:69]:
```suggestion
                <Peers>
                  <Peer Address="127.0.0.1"/>
                </Peers>
              </Discovery>

File: .github/workflows/workspace_integration_test.yaml [hunk 5]
```diff
@@ -1,12 +1,107 @@
...
+        with:
+          submodules: recursive
+      # The 'lo' interfaces will not have multicast, and will fail trying to claim a participant index
+      # with the default cyclone dds config. We set one for the unit tests to be able to execute.
+      - name: Configure DDS
+        run: |
+          cat <<EOF > /tmp/cyclonedds.xml
+          <?xml version="1.0" ?>
+          <CycloneDDS xmlns="https://cdds.io/config" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://cdds.io/config https://raw.githubusercontent.com/eclipse-cyclonedds/cyclonedds/master/etc/cyclonedds.xsd">
+            <Domain id="any">
```
Comment (sea-bass) [inline .github/workflows/workspace_integration_test.yaml:65]:
```suggestion
            <Domain id="any">
                      <General>
            <Interfaces>
                <NetworkInterface autodetermine="false" name="lo" />
            </Interfaces>
            <AllowMulticast>false</AllowMulticast>
        </General>

**Comment by sea-bass** [comment]
I think this one is good to merge!

**Pull request merged**